docs/user-guide/arrays.md

```diff
@@ -566,6 +566,124 @@ In this example a shard shape of (1000, 1000) and a chunk shape of (100, 100) is
+(totaling 60), and the second dimension is divided into 4 chunks of size 25 (totaling 100).
+
+### Reading and writing
+
+Arrays with variable chunking support the same read/write operations as regular arrays:
+
+```python exec="true" session="arrays" source="above" result="ansi"
+# Write data
+data = np.arange(60 * 100, dtype='float32').reshape(60, 100)
+z[:] = data
+
+# Read data back
+result = z[:]
+print(f"Data matches: {np.all(result == data)}")
+print(f"Slice [10:30, 50:75]: {z[10:30, 50:75].shape}")
+```
+
+### Accessing chunk information
+
+With variable chunking, the standard `.chunks` property is not available since chunks
+have different sizes. Instead, access chunk information through the chunk grid:
+
+```python exec="true" session="arrays" source="above" result="ansi"
+from zarr.core.chunk_grids import RectilinearChunkGrid
+
+# Access the chunk grid
+chunk_grid = z.metadata.chunk_grid
+print(f"Chunk grid type: {type(chunk_grid).__name__}")
+
+# Get chunk shapes for each dimension
+if isinstance(chunk_grid, RectilinearChunkGrid):
+    print(f"Dimension 0 chunk sizes: {chunk_grid.chunk_shapes[0]}")
+    print(f"Dimension 1 chunk sizes: {chunk_grid.chunk_shapes[1]}")
+    print(f"Total number of chunks: {chunk_grid.get_nchunks((60, 100))}")
+```
+
+### Use cases
+
+Variable chunking is particularly useful for:
+
+1. **Irregular time series**: When your data has non-uniform time intervals, you can
+   create chunks that align with your sampling periods.
+
+2. **Aligning with partitions**: When you need to match chunk boundaries with existing
+   data partitions or structural boundaries in your data.
+
+3. **Optimizing access patterns**: When certain regions of your array are accessed more
+   frequently, you can use smaller chunks there for finer-grained access.
+
+### Example: Time series with irregular intervals
+
+```python exec="true" session="arrays" source="above" result="ansi"
+# Daily measurements for one year, chunked by month
+# Each chunk corresponds to one month (varying from 28-31 days)
+z_timeseries = zarr.create_array(
+    store='data/example-22.zarr',
+    shape=(365, 100),  # 365 days, 100 measurements per day
+    chunks=[[31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], [100]],  # Days per month
+    dtype='float64',
+    zarr_format=3
+)
+print(f"Created array with shape {z_timeseries.shape}")
+print(f"Chunk shapes: {z_timeseries.metadata.chunk_grid.chunk_shapes}")
+print(f"Number of chunks: {len(z_timeseries.metadata.chunk_grid.chunk_shapes[0])} months")
+```
+
+### Limitations
+
+Variable chunking has some important limitations:
+
+1. **Zarr v3 only**: This feature is only available when using `zarr_format=3`.
+   Attempting to use variable chunks with `zarr_format=2` will raise an error.
+
+2. **Not compatible with sharding**: You cannot use variable chunking together with
+   the sharding feature. Arrays must use either variable chunking or sharding, but not both.
```

Comment:
I hope this is a temporary limitation! There's a natural extension of rectilinear chunk grids to rectilinear shard grids.

Reply:
xref: https://github.com/zarr-developers/zarr-extensions/issues/34